netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -365,20 +366,16 @@ void overrideAuthorityChecker(@Nullable OverrideAuthorityChecker authorityChecke
   @VisibleForTesting
   @CheckReturnValue
   static ProtocolNegotiator createProtocolNegotiator(
+      SocketAddress targetServerAddress,
       String authority,
       NegotiationType negotiationType,
-      SslContext sslContext) {
+      SslContext sslContext
+  ) {
     ProtocolNegotiator negotiator =
         createProtocolNegotiatorByType(authority, negotiationType, sslContext);
-    String proxy = System.getenv("GRPC_PROXY_EXP");
+    SocketAddress proxy = Proxies.proxyFor(targetServerAddress);
     if (proxy != null) {
-      String[] parts = proxy.split(":", 2);
-      int port = 80;
-      if (parts.length > 1) {
-        port = Integer.parseInt(parts[1]);
-      }
-      InetSocketAddress proxyAddress = new InetSocketAddress(parts[0], port);
-      negotiator = ProtocolNegotiators.httpProxy(proxyAddress, null, null, negotiator);
+      negotiator = ProtocolNegotiators.httpProxy(proxy, null, null, negotiator);
```

Comment:
ProxySelector assumes you will be using Authenticator as well. We can break setups if we use ProxySelector without using Authenticator.